src/__experimental__/components/checkbox/checkbox-group.component.js

```diff
@@ -7,26 +7,54 @@ import FormField from '../form-field';
 
 const CheckboxGroup = (props) => {
   const {
+    id,
     children,
-    groupName,
     hasError,
     hasWarning,
-    hasInfo
+    hasInfo,
+    onChange,
+    onBlur,
+    value
   } = props;
 
-  const groupLabelId = `${groupName}-label`;
+  const checkboxGroupValue = (value !== undefined) ? value : [];
+  const groupLabelId = `${id}-label`;
+  const [checkedValue, setCheckedValue] = useState(checkboxGroupValue);
 
-  const buttons = React.Children.map(children, (child) => {
-    const handleChange = (ev) => {
-      child.props.onChange(ev);
+  const onChangeProp = useCallback((e) => {
+    const checkedCheckboxes = [...checkedValue];
+    const checkedIndex = checkedCheckboxes.indexOf(e.target.value);
+
+    if (checkedIndex === -1) {
+      checkedCheckboxes.push(e.target.value);
+    } else {
+      checkedCheckboxes.splice(checkedIndex, 1);
+    }
+
+    setCheckedValue(checkedCheckboxes);
+
+    const name = (id && { id });
+
+    e.target = {
+      ...(id && { id }),
+      ...name,
+      value: checkedCheckboxes
     };
 
+    onChange(e);
+  }, [onChange, checkedValue]);
+
+  const buttons = React.Children.map(children, (child) => {
+    const checked = (checkedValue.indexOf(child.props.value) !== -1);
+
```

Comment:
This solution does not support a `defaultChecked` prop passed to the one of elements in case of uncontrolled use of `CheckboxGroup`

Reply:
added `defaultValue` prop.